src/codemagic/tools/_app_store_connect/actions/publish_action.py

```diff
@@ -383,6 +396,8 @@ def _process_ipa_after_upload(
                 **testflight_options.__dict__,
                 'max_build_processing_wait': 0,  # Overwrite waiting since we already waited above.
             }
+            if testflight_options.expire_builds:
+                self.expire_builds(application_id=app.id, build_ids=[build.id], should_print=False)
```

Comment:
As there can be at most one build that is currently in review, and we just need to expire this, then I'd say that here we have a serious flaw in logic. We should detect only this single build that needs to be expired, or otherwise canceled. Probably this can be done by adding additional state parameter or something to the `expire_builds` method.